omnij-rpc/src/main/java/foundation/omni/rpc/OmniExtendedClient.java

```diff
@@ -103,10 +104,10 @@ public Sha256Hash acceptDexOffer(Address fromAddress, CurrencyID currencyId, Big
     public Sha256Hash createMetaDexSellOffer(Address address, CurrencyID currencyForSale, BigDecimal amountForSale,
                                       CurrencyID currencyDesired, BigDecimal amountDesired,
                                       Byte action) throws JsonRPCException, IOException {
-        Long willetsForSale = BTC.btcToSatoshis(amountForSale).longValue();  // Assume divisible property
-        Long willetsDesired = BTC.btcToSatoshis(amountDesired).longValue();  // Assume divisible property
+        Coin willetsForSale = BTC.btcToCoin(amountForSale);  // Assume divisible property
```

Comment:
:)

L83 should also be changed to `OmniDivisibleValue`.

So just to summarize:

createDexSellOffer:

- L64: `amountForSale` should be `OmniDivisibleValue` (it's always either MSC or TMSC)
- L65: `amountDesired` can remain as `Coin` (it's always BTC)
- L66:  `commitmentFee` can remain as `Coin` (it's always BTC)

acceptDexOffer:

- L84: `amount` amount should be `OmniDivisibleValue` (it's always either MSC or TMSC)

createMetaDexSellOffer:

- L107: `amountForSale` should be `OmniDivisibleValue` for now (it could be divisible or indivisble in the future)
- L108: `amountDesired` should be `OmniDivisibleValue` for now (it could be divisible or indivisble in the future)